api/src/Appointments/Doctrine/AppointmentRepairerExtension.php

```diff
@@ -35,9 +36,13 @@ private function addWhere(QueryBuilder $queryBuilder, string $resourceClass): vo
         }
 
         $rootAlias = $queryBuilder->getRootAliases()[0];
-        $queryBuilder
-            ->andWhere(sprintf('%s.repairer = :repairerId', $rootAlias))
-            ->setParameter('repairerId', $user->repairer ? $user->repairer->id : ($user->repairerEmployee?->repairer->id))
-        ;
+        $repairers = $user->repairers->count() > 0 ? $user->repairers->toArray() : [$user->repairerEmployee->repairer];
+        $repairerIds = array_map(function (Repairer $repairer): int {
+            return $repairer->id;
+        }, $repairers);
+
+        $queryBuilder->innerJoin(sprintf('%s.repairer', $rootAlias), 'r')
+            ->orWhere('r.id IN (:repairersIds)')
+            ->setParameter('repairersIds', $repairerIds);
     }
 }
```

Comment:
That would be nice to add a test with a owner (boss) having single or multiple repairers, a test when there is this case (employee ?) :
`$user->repairerEmployee->repairer`
and a test for roles ADMIN/USER to test the condition